src/main/java/com/product/controller/ProductController.java

```diff
@@ -0,0 +1,53 @@
+package com.product.controller;
+
+import java.util.List;
+
+import javax.validation.Valid;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.product.model.Product;
+import com.product.service.ProductServiceImpl;
+
+@RestController
+@RequestMapping("/api")
+public class ProductController {
+
+	private ProductServiceImpl productServiceImpl;
+
+	@Autowired
+	public ProductController(ProductServiceImpl productServiceImpl) {
+		this.productServiceImpl = productServiceImpl;
+	}
+
+	@GetMapping("/products")
+	public List<Product> getProduct() {
+		return productServiceImpl.getAllProducts();
+	}
+
+	@PostMapping("/products")
+	public String createProduct(@Valid @RequestBody Product product) {
```

Comment:
When I meant return String, I meant to not do `ResponseEntity<Object>` but do `ResponseEntity<String>` instead.

Please always return `ResponseEntity` so that it goes out as a standard response, and you can send HTTP codes as well. Eg createProducts should return Http Code 201, get should return Http code 200 etc

Change it . for all the APIs

Reply:
Done.